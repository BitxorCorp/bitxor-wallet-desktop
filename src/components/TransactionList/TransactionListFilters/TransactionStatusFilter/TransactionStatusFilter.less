@import '../../../../views/resources/css/variables.less';

.transactions-filter {
    position: relative;
    color: @primary;
}

.filter-button {
    position: relative;
    box-shadow: unset;
    border: unset;
    background: @grayLightest;
    border-radius: 3.4px;
    padding: 0.07rem 0.12rem;
    font-size: @smallFont;
    cursor: pointer;
}

.filter-button-arrow {
    position: absolute;
    top: 45%;
    right: 0.12rem;
    line-height: 1;
    transform: translateY(-50%) rotate(135deg);
    font-size: 0.14rem;
    color: #808695;
    width: 0.07rem;
    height: 0.07rem;
    border-top: 1px solid @primary;
    border-right: 1px solid @primary;
    transition: all 0.2s ease-in-out;
    display: inline-block;
    font-family: Ionicons;
    speak: none;
    font-style: normal;
    font-weight: 400;
    font-variant: normal;
    text-transform: none;
    text-rendering: optimizeLegibility;
    -webkit-font-smoothing: antialiased;
    -moz-osx-font-smoothing: grayscale;
    vertical-align: -0.125em;
    text-align: center;
}

.arrow-up {
    top: 55%;
    transform: translateY(-50%) rotate(315deg);
    display: inline-block;
}

.filter-options-container {
    position: absolute;
    top: calc(@smallerFont + 0.14rem);
    width: 100%;
    background: @grayLightest;
    font-size: @smallFont;
    z-index: 1061 !important;
    padding: 0.14rem 0;

    &__option {
        margin-bottom: 0.07rem;
        padding: 0 0.15rem;
    }

    &__divider {
        width: 100%;
        height: 1px;
        background: @grayLight;
        margin: 0.14rem 0 0.07rem 0;
    }
}

/deep/ .ivu-checkbox {
    .ivu-checkbox-inner {
        background: white;
        box-shadow: 0 0.02rem 0.04rem rgba(37, 46, 55, 0.0774148);
    }
}

/deep/ .ivu-checkbox-checked {
    border-radius: 0.02rem;
   // background: @primary !important;

    .ivu-checkbox-inner {
        background: transparent;
    }
}
