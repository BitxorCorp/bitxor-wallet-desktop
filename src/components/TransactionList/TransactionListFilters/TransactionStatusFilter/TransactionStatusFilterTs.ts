import { Component, Vue, Watch } from 'vue-property-decorator';
import { mapGetters } from 'vuex';
import { Signer } from '@/store/Account';
import { FilterOption, TransactionFilterOptions } from '@/store/Transaction';
import { Checkbox } from 'view-design';

@Component({
    computed: {
        ...mapGetters({
            currentSigner: 'account/currentSigner',
        }),
    },
    components: {
        Checkbox,
    },
})
export class TransactionStatusFilterTs extends Vue {
    public currentSigner: Signer;
    public isSelectionShown: boolean = false;
    public transactionFilterOptions = TransactionFilterOptions;

    public onChange(key, value): void {
        this.$store.commit('transaction/filterTransactions', {
            filterOption: new FilterOption(key, value),
            currentSignerAddress: this.currentSigner.address.plain(),
        });
    }

    /**
     * Toggles selection options block.
     */
    public toggleSelection(): void {
        this.isSelectionShown = !this.isSelectionShown;
    }

    /**
     * Closes selection options block.
     */
    public closeSelection(): void {
        this.isSelectionShown = false;
    }

    @Watch('currentSigner')
    onCurrentSignerChange() {
        this.$store.commit('transaction/filterTransactions', {
            filterOption: null,
            currentSignerAddress: this.currentSigner.address.plain(),
        });
    }
}
