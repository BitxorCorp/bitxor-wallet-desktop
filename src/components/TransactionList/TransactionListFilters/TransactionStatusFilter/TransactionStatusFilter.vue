<template>
    <div class="transactions-filter">
        <div class="filter-button" @click.stop="toggleSelection">
            <span>Filter</span>
            <div class="filter-button-arrow" :class="{ 'arrow-up': isSelectionShown }" />
        </div>
        <div v-show="isSelectionShown" v-click-outside="closeSelection" class="filter-options-container">
            <div class="filter-options-container__option">
                <Checkbox id="Confirmed" @input="(value) => onChange(transactionFilterOptions.confirmed, value)" />
                <label for="Confirmed">Confirmed</label>
            </div>
            <div class="filter-options-container__option">
                <Checkbox id="Unconfirmed" @input="(value) => onChange(transactionFilterOptions.unconfirmed, value)" />
                <label for="Unconfirmed">Unconfirmed</label>
            </div>
            <div class="filter-options-container__option">
                <Checkbox id="Partial" @input="(value) => onChange(transactionFilterOptions.partial, value)" />
                <label for="Partial">Partial</label>
            </div>
            <div class="filter-options-container__divider"></div>
            <div class="filter-options-container__option">
                <Checkbox id="Sent" @input="(value) => onChange(transactionFilterOptions.sent, value)" />
                <label for="Sent">Sent</label>
            </div>
            <div class="filter-options-container__option">
                <Checkbox id="Received" @input="(value) => onChange(transactionFilterOptions.received, value)" />
                <label for="Received">Received</label>
            </div>
        </div>
    </div>
</template>
<script lang="ts">
// @ts-ignore
import { TransactionStatusFilterTs } from './TransactionStatusFilterTs';
export default class TransactionStatusFilter extends TransactionStatusFilterTs {}
</script>
<style lang="less" scoped>
@import './TransactionStatusFilter.less';
</style>
