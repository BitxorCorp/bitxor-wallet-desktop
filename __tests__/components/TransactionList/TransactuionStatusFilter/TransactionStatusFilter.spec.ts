//@ts-ignore
import TransactionStatusFilter from '@/components/TransactionList/TransactionListFilters/TransactionStatusFilter/TransactionStatusFilter.vue';
import { getComponent } from '@MOCKS/Components';
import AccountStore from '@/store/Account';
import TransactionStore from '@/store/Transaction';
import { Vue } from 'vue-property-decorator';
let wrapper;
/* eslint-disable @typescript-eslint/no-unused-vars */
let vm;
beforeEach(() => {
    wrapper = getComponent(
        TransactionStatusFilter,
        { account: AccountStore, transaction: TransactionStore },
        { currentAccount: null, signers: [] },
        {},
        {},
    );
    vm = wrapper.vm as TransactionStatusFilter;
});
afterEach(() => {
    wrapper.destroy();
});
describe('TransactionStatusFilter', () => {
    let wrapper: any;
    // eslint-disable-next-line @typescript-eslint/no-unused-vars
    let vm: any;
    beforeEach(() => {
        wrapper = getComponent(
            TransactionStatusFilter,
            { account: AccountStore, transaction: TransactionStore },
            { currentAccount: null, signers: [] },
            {},
            {},
        );
        vm = wrapper.vm as TransactionStatusFilter;
    });
    afterEach(() => {
        wrapper.destroy();
    });
    describe('TransactionStatusFilter', () => {
        test('renders correctly', async (): Promise<void> => {
            expect(wrapper).toMatchSnapshot();

            wrapper.find('.filter-button').trigger('click');

            await Vue.nextTick();

            expect(wrapper).toMatchSnapshot();
        });
    });
});
