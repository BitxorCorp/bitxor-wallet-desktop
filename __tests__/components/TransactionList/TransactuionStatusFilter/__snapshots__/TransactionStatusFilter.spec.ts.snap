// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransactionStatusFilter TransactionStatusFilter renders correctly 1`] = `
<div class="transactions-filter">
  <div class="filter-button"><span>Filter</span>
    <div class="filter-button-arrow"></div>
  </div>
  <div class="filter-options-container" style="display: none;">
    <div class="filter-options-container__option">
      <checkbox-stub truevalue="true" size="default" id="Confirmed"></checkbox-stub> <label for="Confirmed">Confirmed</label>
    </div>
    <div class="filter-options-container__option">
      <checkbox-stub truevalue="true" size="default" id="Unconfirmed"></checkbox-stub> <label for="Unconfirmed">Unconfirmed</label>
    </div>
    <div class="filter-options-container__option">
      <checkbox-stub truevalue="true" size="default" id="Partial"></checkbox-stub> <label for="Partial">Partial</label>
    </div>
    <div class="filter-options-container__divider"></div>
    <div class="filter-options-container__option">
      <checkbox-stub truevalue="true" size="default" id="Sent"></checkbox-stub> <label for="Sent">Sent</label>
    </div>
    <div class="filter-options-container__option">
      <checkbox-stub truevalue="true" size="default" id="Received"></checkbox-stub> <label for="Received">Received</label>
    </div>
  </div>
</div>
`;

exports[`TransactionStatusFilter TransactionStatusFilter renders correctly 2`] = `
<div class="transactions-filter">
  <div class="filter-button"><span>Filter</span>
    <div class="filter-button-arrow arrow-up"></div>
  </div>
  <div class="filter-options-container" style="">
    <div class="filter-options-container__option">
      <checkbox-stub truevalue="true" size="default" id="Confirmed"></checkbox-stub> <label for="Confirmed">Confirmed</label>
    </div>
    <div class="filter-options-container__option">
      <checkbox-stub truevalue="true" size="default" id="Unconfirmed"></checkbox-stub> <label for="Unconfirmed">Unconfirmed</label>
    </div>
    <div class="filter-options-container__option">
      <checkbox-stub truevalue="true" size="default" id="Partial"></checkbox-stub> <label for="Partial">Partial</label>
    </div>
    <div class="filter-options-container__divider"></div>
    <div class="filter-options-container__option">
      <checkbox-stub truevalue="true" size="default" id="Sent"></checkbox-stub> <label for="Sent">Sent</label>
    </div>
    <div class="filter-options-container__option">
      <checkbox-stub truevalue="true" size="default" id="Received"></checkbox-stub> <label for="Received">Received</label>
    </div>
  </div>
</div>
`;
